onnxruntime/core/framework/cloud_invoker.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#pragma once
+
+#ifdef USE_CLOUD
+#include <memory>
+#include "gsl/gsl"
+#include "core/framework/tensor.h"
+#include "core/common/inlined_containers.h"
+
+namespace onnxruntime {
+
+using CloudEndPointConfig = std::unordered_map<std::string, std::string>;
+using TensorPtr = std::unique_ptr<onnxruntime::Tensor>;
+using TensorPtrArray = onnxruntime::InlinedVector<TensorPtr>;
+
+class CloudEndPointInvoker {
+ public:
+  CloudEndPointInvoker(const CloudEndPointConfig& config, AllocatorPtr& allocator);
+  virtual ~CloudEndPointInvoker() = default;
+  ORT_DISALLOW_COPY_ASSIGNMENT_AND_MOVE(CloudEndPointInvoker);
+
+  static Status CreateInvoker(const CloudEndPointConfig& config,
+                              AllocatorPtr allocator,
```

Comment:
Allocator parameter passing needs to be fixed. It's a mixed bag of pass by value and ref. Use const-ref everywhere for uniformity. See https://herbsutter.com/2013/06/05/gotw-91-solution-smart-pointer-parameters/

Reply:
Now passing them all as const ref.